db/service/src/main/java/org/commonjava/indy/db/service/ServiceStoreDataCacheUpdater.java

```diff
@@ -137,22 +135,34 @@ private void obsoleteQueryCache( StoreKey storeKey )
             for ( Map.Entry<Object, Collection<ArtifactStore>> entry : cache.entrySet() )
             {
                 Object key = entry.getKey();
+                Collection<ArtifactStore> affectedGroups = new HashSet<>();
                 // This is for getGroupsAffectedBy query cache
                 if ( key instanceof Set && ( (Set) key ).contains( storeKey ) )
                 {
                     logger.info( "Fresh the store query cache, removed: {}", storeKey );
+                    affectedGroups = cache.get( key );
                     cache.remove( key );
                 }
-                // This is for getOrderedConcreteStoresInGroup query cache
-                if ( storeKey.getType() == StoreType.group && key instanceof String && ( (String) key ).indexOf(
-                        String.format( "%s:%s", storeKey.getPackageType(), storeKey.getName() ) ) > 0 )
+                // This is for affectedGroups' getOrderedConcreteStoresInGroup query cache
+                for ( ArtifactStore group : affectedGroups )
                 {
-                    logger.info( "Fresh the concrete stores query cache, removed: {}", storeKey );
-                    cache.remove( key );
+                    clearOrderedConcreteStoresCache( key, group.getKey(), cache );
```

Comment:
I'm a little curious if this will covered. If the affectedGroups is not empty, that means the current "key" is a set, but this clearOrderedConcreteStoresCache needs the "key" to be string, which will bypass the check.
For the OrderedConcreteStoresCache, its key is a string like "maven:groupname:true:orderedConcreteStoresInGroup", so maybe you need another way to do this clean?

Reply:
@ligangty, yeah, right, I'll update, thanks